### PR TITLE
Include TextureLayer in contents of toImageSync

### PR DESCRIPTION
**Pull request by jonahwilliams**
See revert https://github.com/flutter/engine/pull/35587

Fixes NPE in GetGrContext

### CODE REVIEW

File: shell/common/rasterizer.cc [hunk 1]
```diff
@@ -145,6 +145,10 @@ std::shared_ptr<flutter::TextureRegistry> Rasterizer::GetTextureRegistry() {
   return compositor_context_->texture_registry();
 }
 
+GrDirectContext* Rasterizer::GetGrContext() {
+  return surface_->GetContext();
```
Comment (dnfield) [inline shell/common/rasterizer.cc:149]:
```suggestion
  return surface_ ? surface_->GetContext() : nullptr;
```

Comment (jonahwilliams) [reply shell/common/rasterizer.cc]: nullsafety for c++ when?

**Comment by jonahwilliams** [comment]
This is almost ready again, but I need to do another local engine build and verify the test now passes

File: shell/common/rasterizer.cc (hunk 1 above)
Comment (dnfield) [reply shell/common/rasterizer.cc]: Something something billion dollar mistake

**Review by dnfield — APPROVED** [review]
LGTM as long as the internal test is passing.

**Comment by jonahwilliams** [comment]
It passed: http://sponge2/ad365424-d3c9-485e-b628-98f0b80d8686

:D

**Pull request merged**